src/Data/Managers/V2/GetInstanceWithRelations/useGetInstanceWithRelations.ts

```diff
@@ -0,0 +1,82 @@
+import { useQuery } from "@tanstack/react-query";
+import { ServiceInstanceModel } from "@/Core";
+import { PrimaryBaseUrlManager } from "@/UI";
+import { InstanceWithReferences } from "../../GetInstanceWithRelations/interface";
+import { useFetchHelpers } from "../helpers";
+
+/**
+ * Custom hook to fetch an instance with its related instances from the API.
+ * @param id - The ID of the instance to fetch.
+ * @param environment - The environment in which we are looking for instances.
+ * @returns An object containing a custom hook to fetch the instance with its related instances.
+ */
+export const useGetInstanceWithRelations = (
+  id: string,
+  environment: string,
+) => {
+  //extracted headers to avoid breaking rules of Hooks
+  const { createHeaders, handleErrors } = useFetchHelpers();
+  const headers = createHeaders(environment);
+  const baseUrlManager = new PrimaryBaseUrlManager(
+    globalThis.location.origin,
+    globalThis.location.pathname,
+  );
+  const baseUrl = baseUrlManager.getBaseUrl(process.env.API_BASEURL);
+
+  /**
+   * Fetches a service instance with its related instances recursively.
+   * @param instanceId - The ID of the instance to fetch.
+   * @returns An object containing the fetched instance and its related instances.
+   * @throws Error if the instance fails to fetch.
+   */
+  const fetchService = async (instanceId: string) => {
```

Comment:
If you make instanceId optional, and fallback on the id passed to the hook, you don't need that extra method on the line 65. 

```suggestion
  const fetchService = async (instanceId: string = id) => {
```

And on line 50 you can keep:  
`const nestedInstance = await fetchService(relatedId);` and it will override the id in the next recursion.